src/hipscat_import/catalog/margin_cache.py

```diff
@@ -0,0 +1,249 @@
+
+    if len(margin_data):
+        # TODO: this should be a utility function in `hipscat`
+        # that properly handles the hive formatting
+        # generate a file name for our margin shard
+        partition_file = paths.pixel_catalog_file(output_path, order, pix)
+        partition_dir = f"{partition_file[:-8]}/"
+        shard_dir = paths.pixel_directory(
+            partition_dir, source_order, source_pix
+        )
+
+        file_io.make_directory(shard_dir, exist_ok=True)
+
+        shard_path = paths.pixel_catalog_file(
+            partition_dir, source_order, source_pix
+        )
+
+        margin_data.to_parquet(shard_path)
+
+def _margin_filter_polar(
+    data,
+    order,
+    pix,
+    margin_order,
+    pole,
+    margin_threshold,
+    ra_column,
+    dec_column,
+    bounding_polygons
+):
+    """Filter out margin data around the poles."""
+    trunc_pix = pixel_math.get_truncated_margin_pixels(
+        order=order,
+        pix=pix,
+        margin_order=margin_order
+    )
+    data["is_trunc"] = np.isin(data["margin_pixel"], trunc_pix)
+
+    # pylint: disable=singleton-comparison
+    trunc_data = data.loc[data["is_trunc"] == True]
+    other_data = data.loc[data["is_trunc"] == False]
+    # pylint: enable=singleton-comparison
+
+    trunc_data["margin_check"] = pixel_math.check_polar_margin_bounds(
+        trunc_data[ra_column].values,
+        trunc_data[dec_column].values,
+        order,
+        pix,
+        margin_order,
+        pole,
+        margin_threshold
+    )
+    other_data["margin_check"] = pixel_math.check_margin_bounds(
+        other_data[ra_column].values,
+        other_data[dec_column].values,
+        bounding_polygons
+    )
+
+    return pd.concat([trunc_data, other_data])
+
+def generate_margin_cache(args):
+    """Generate a margin cache for a given input catalog.
+    The input catalog must be in hipscat format.
+    This method will handle the creation of the `dask.distributed` client
+    based on the `dask_tmp`, `dask_n_workers`, and `dask_threads_per_worker`
+    values of the `MarginCacheArguments` object.
+
+    Args:
+        args (MarginCacheArguments): A valid `MarginCacheArguments` object.
+    """
+    with Client(
+        local_directory=args.dask_tmp,
+        n_workers=args.dask_n_workers,
+        threads_per_worker=args.dask_threads_per_worker,
+    ) as client:  # pragma: no cover
+        generate_margin_cache_with_client(
+            client,
+            args
+        )
+
+def generate_margin_cache_with_client(client, args):
+    """Generate a margin cache for a given input catalog.
+    The input catalog must be in hipscat format.
+    Args:
+        client (dask.distributed.Client): A dask distributed client object.
+        args (MarginCacheArguments): A valid `MarginCacheArguments` object.
+    """
+    # determine which order to generate margin pixels for
+    partition_stats = args.catalog.get_pixels()
```

Comment:
Hmm.. Catalog should probably expose the healpixPixels list instead/as well, so users don't need to know the column names all the time.

https://github.com/astronomy-commons/hipscat/blob/main/src/hipscat/catalog/partition_info.py#L21

Reply:
Oh. You're using it below as `args.catalog.partition_info.get_healpix_pixels()`. Awesome. This should probably use that too for consistency, and so as not to use extra column name dependencies when you don't have to.